internal/config/types.go

```diff
@@ -123,17 +125,63 @@ func (c *Config) HasSecret(secret string) bool {
 type Environment = map[string]string
 
 type Python struct {
-	Version        string `toml:"version" json:"version"`
+	Version        string `toml:"version,omitempty" json:"version"`
 	PackageFile    string `toml:"package_file,omitempty" json:"packageFile"`
 	PackageManager string `toml:"package_manager,omitempty" json:"packageManager"`
 }
 
+func (p *Python) FillDefaults(
+	pythonInterpreter *interpreters.PythonInterpreter,
+) {
+	if p != nil && pythonInterpreter != nil && (*pythonInterpreter).IsPythonExecutableValid() {
+		python := *pythonInterpreter
+		if p.Version == "" {
+			pythonVersion, pythonVersionError := python.GetPythonVersion()
```

Comment:
with the change suggested above, this will get easier: ~python := *pythonInterpreter~
```
if p.Version == "" {
	pythonVersion, pythonVersionError := pythonInterpreter.GetPythonVersion()
```

All these comments apply for R below too.

Reply:
Updated for python as well as R.